### PR TITLE
add backend/obj.d

### PR DESCRIPTION
**Pull request by WalterBright**
This is a conversion of obj.h to D, part of a continuing conversion of all of the DMD front end to D. It requires some adjustment to the other C++ files.

### CODE REVIEW

**Comment by WalterBright** [comment]
Blocking https://github.com/dlang/dmd/pull/6138

**Comment by WalterBright** [comment]
ah, wonderful!

File: src/backend/machobj.c [hunk 1]
```diff
@@ -2861,10 +2861,6 @@ symbol *Obj::tlv_bootstrap()
     return tlv_bootstrap_sym;
 }
 
-MachObj64::MachObj64()
-{
-    assert(I64);
-}
```
Comment (jacob-carlborg) [inline src/backend/machobj.c]: Why was this removed?

Comment (WalterBright) [inline src/backend/machobj.c]: Because of problems calling constructors across language boundaries.

Comment (jacob-carlborg) [inline src/backend/machobj.c]: Ah, I see, fari enough. So it's not possible to have a constructor implemented in D and call it from C++, but the default constructor works?
